File: data_structures/arrays/arrays_ds.py
import math


def reverseArray(array):
    n = len(array)
    last = n - 1
    middle = math.floor(n / 2)

    # + 1 because range excludes the upper bound
    for left in range(middle):
        right = last - left
        temp = array[left]
        array[left] = array[right]
        array[right] = temp

    return array


if __name__ == "__main__":
    input_data = [1, 4, 3, 2]
    print(reverseArray(input_data))

File: hackerrank/data_structures/arrays/arrays_ds.java
package data_structures.arrays;

public class arrays_ds {
  static int[] reverseArray(int[] array) {
    int n = array.length;
    int last = n - 1;
    int middle = Math.floorDiv(n, 2);

    for(int left = 0; left < middle; left++) {
      int right = last - left;
      int temp = array[left];
      array[left] = array[right];
      array[right] = temp;
    }
    return array;
  }
  
  public static void main(String[] args) {
    int[] inputData = {1,4,3,2};
    int[] output = reverseArray(inputData);
    for(int item: output) {
      System.out.println(item);
    }
  }
}